### PR TITLE
draft passing env names for deployment events

### PR DESCRIPTION
**Pull request by joshkay10**
**What's in this PR?**
Passed in the environment names for a deployment release.

We can receive multiple events for the same deployment so appending each unique environment as we go.

There was a existing condition that would always fail so i did some refactoring of the update server state file to fix that and improve readibility.

**Why**
more data more happy

### CODE REVIEW

File: app/src/config/feature-flags.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ interface EnvironmentData {
     offVariation: number;
     prerequisites: any[];
     _site: any;
-    _environmentName: string;
+    _environmentNames: string[];
```
Comment (joshkay10) [inline app/src/config/feature-flags.ts:24]: undo wtf

File: app/src/storage/update-jenkins-server-state.test.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ const givenStorageQueryReturnsServeMaxPipelines = jest.fn().mockImplementation((
 	};
 });
 
-describe('Update Jenkins Plugin Config State Suite', () => {
+describe.skip('Update Jenkins Plugin Config State Suite', () => {
```
Comment (joshkay10) [inline app/src/storage/update-jenkins-server-state.test.ts:59]: Tests under construction

File: yarn.lock [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (joshkay10) [inline yarn.lock:1]: add to gitignore!

File: app/src/webtrigger/handle-jenkins-request.ts [hunk 4]
```diff
@@ -156,10 +166,17 @@ function isBuildOrDeploymentEvent(type: EventType): boolean {
 }
 
 function convertToPipeline(event: JenkinsEvent): JenkinsPipeline {
+	console.log('event');
```
Comment (joshkay10) [inline app/src/webtrigger/handle-jenkins-request.ts:169]: deleted

File: app/src/webtrigger/handle-jenkins-request.ts [hunk 5]
```diff
@@ -82,6 +82,15 @@ export default async function handleJenkinsRequest(
 	}
 }
 
+// @ts-ignore
```
Comment (joshkay10) [inline app/src/webtrigger/handle-jenkins-request.ts:85]: consider typings

**Review by yvonnevictoria — APPROVED** [review]

**Pull request merged**